src/main/java/mate/academy/dao/MovieDaoImpl.java

```diff
@@ -0,0 +1,48 @@
+package mate.academy.dao;
+
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+
+    @Override
+    public Movie add(Movie movie) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("movie didn't save" + e.getMessage());
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return movie;
+    }
+
+    @Override
+    public Movie get(Long id) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()){
+            return session.get(Movie.class, id);
+        } catch (Exception e) {
+            throw new NoSuchElementException("Movie not found with id: " + id);
```

Comment:
```suggestion
            throw new DataProcessingException("Movie not found with id: " + id);
```

let`s throw DataProcessingException here(in case the error occurred not due to the absence of an entity in the database)